sourcegraph/jetbrains
# Cody Ignore: Ignore Chat UX updates

- @-mention selector for files badges ignored files
- Context inspection summary displays elaborate count of ignored files
- Context inspection disclosure displays ignored files with strikeout
- Repo picker displays ignored repos with a slashed circle, tooltips, etc.

This also changes the enhanced context selector to reflect the state of the extension. This has many effects:

- The implicitly included repository is displayed (adds tooltips, etc. for this.)
- Removing repos is done by clicking the checkboxes, and they disappear. This needs UX feedback. It is close to VSCode, but it is weird to have the items disappear.
- There's a bug in Agent where the git extension shim doesn't initialize the repository `state.remotes`, causing us to pick up the implicit repo poorly.

Part of #1256

## Test plan

![Screenshot 2024-05-07 at 23 12 17](https://github.com/sourcegraph/jetbrains/assets/55120/66b8fd15-1f1b-4761-8ac9-051b73e5f86c)

- Sign in to sg02.sourcegraphcloud.com
- Open the Cody repo, add a bunch of repos and verify the repos have a circle-slash in line with the sg02 policy
- Mention files with @ and the picker should say the files are ignored
- Sending a chat with ignored mentioned files and the context disclosure should display the files with strikeout text
